File: src/simulation_control/src/circle_driver.cpp
#include <ros/ros.h>
#include <geometry_msgs/Twist.h>
#include <stdlib.h>
#include <time.h>

int main(int argc, char **argv) {
  ros::init(argc, argv, "husky_controller");
  ros::NodeHandle nh;
  ros::Publisher pub = nh.advertise<geometry_msgs::Twist>("husky_velocity_controller/cmd_vel", 100);
  ros::Rate rate(10);

  int start, current, elapsed;
  start = time(0);

  while(ros::ok()) {
    geometry_msgs::Twist msg;
    current = time(0);
    elapsed = current - start;
    
    if(elapsed <= 10){
      msg.linear.x = 0.0;
      msg.angular.z = -2.0;
      pub.publish(msg);
    }
    else if(elapsed > 10 && elapsed <= 20){
      msg.linear.x = 0.0;
      msg.angular.z = 2.0;
      pub.publish(msg);
    }
    else{
      msg.linear.x = 0.0;
      msg.angular.z = 0.0;
      pub.publish(msg);
    }
    
    rate.sleep();
  }
}

File: src/simulation_control/src/circle_driver.py
#!/usr/bin/env python

import rospy
import time
from geometry_msgs.msg import Twist


def move():
    rospy.init_node('circle_driver', anonymous=True)
    vel_pub = rospy.Publisher('husky_velocity_controller/cmd_vel', Twist, queue_size=100)
    vel_msg = Twist()

    rate = rospy.Rate(10)

    start_time = time.time()

    while not rospy.is_shutdown():
        current_time = time.time()
        elapsed_time = current_time - start_time

        if elapsed_time <= 10.0:
            vel_msg.linear.x = 0.0
            vel_msg.angular.z = -2.0
            vel_pub.publish(vel_msg)

        elif elapsed_time > 10.0 and elapsed_time <= 20.0:
            vel_msg.linear.x = 0.0
            vel_msg.angular.z = 2.0
            vel_pub.publish(vel_msg)

        else:
            vel_msg.linear.x = 0.0
            vel_msg.angular.z = 0.0
            vel_pub.publish(vel_msg)

        rate.sleep()

if __name__ == '__main__':
    try:
        move()
    except rospy.ROSInterruptException: 
        pass